### PR TITLE
[Workflow] Add Python samples to patterns doc

### PR DESCRIPTION
**Pull request by cgillum**
Thank you for helping make the Dapr documentation better!

**Please follow this checklist before submitting:**
- [x] Commits are signed with Developer Certificate of Origin (DCO - [learn more](https://docs.dapr.io/contributing/contributing-overview/#developer-certificate-of-origin-signing-your-work))
- [x] [Read the contribution guide](https://docs.dapr.io/contributing/docs-contrib/contributing-docs/)
- [x] Commands include options for Linux, MacOS, and Windows within codetabs
- [x] New file and folder names are globally unique
- [x] Page references use shortcodes instead of markdown or URL links
- [x] Images use HTML style and have alternative text
- [x] Places where multiple code/command options are given have codetabs

In addition, please fill out the following to help reviewers understand this pull request:

## Description

This PR updates the workflow patterns doc with Python code examples.

## Issue reference

Closes https://github.com/dapr/docs/issues/3496

### CODE REVIEW

**Comment by cgillum** [comment]
@msfussell @hhunter-ms this is ready for review. However, I wasn't able to view it locally due to the following error:

```bash
> hugo server
Start building sites … 
hugo v0.110.0-e32a493b7826d02763c3b79623952e625402b168+extended windows/amd64 BuildDate=2023-01-17T12:16:09Z VendorInfo=gohugoio
Error: Error building site: "C:\GitHub\dapr-docs\daprdocs\content\en\_index.md:10:1": failed to extract shortcode: template for shortcode "alert" not found
Built in 1779 ms
```

Is there a known problem with the docs build? I noticed the link validator in the CI also isn't working but that seems unrelated.

**Comment by greenie-msft** [comment]
> @msfussell @hhunter-ms this is ready for review. However, I wasn't able to view it locally due to the following error:
> 
> ```shell
> > hugo server
> Start building sites … 
> hugo v0.110.0-e32a493b7826d02763c3b79623952e625402b168+extended windows/amd64 BuildDate=2023-01-17T12:16:09Z VendorInfo=gohugoio
> Error: Error building site: "C:\GitHub\dapr-docs\daprdocs\content\en\_index.md:10:1": failed to extract shortcode: template for shortcode "alert" not found
> Built in 1779 ms
> ```
> 
> Is there a known problem with the docs build? I noticed the link validator in the CI also isn't working but that seems unrelated.

There is an issue but we suspect it's related to the version of mistune that was updated recently and is a dependency of Mechanical Markdown (link validation tool).

I suspect the error you're encountering is unrelated? Looking into it now.

**Review by hhunter-ms — APPROVED** [review]
lgtm

File: daprdocs/content/en/developing-applications/building-blocks/workflow/workflow-patterns.md [hunk 1]
```diff
@@ -56,14 +56,61 @@ catch (TaskFailedException) // Task failures are surfaced as TaskFailedException
 }
 ```
 
+{{% alert title="Note" color="primary" %}}
```
Comment (msfussell) [inline daprdocs/content/en/developing-applications/building-blocks/workflow/workflow-patterns.md:59]: Should this note also be in the Python example? It has the same brevity?

File: daprdocs/content/en/developing-applications/building-blocks/workflow/workflow-patterns.md [hunk 2]
```diff
@@ -56,14 +56,61 @@ catch (TaskFailedException) // Task failures are surfaced as TaskFailedException
...
+
+
+def error_handler(ctx, error):
+    print(f'Executing error handler: {error}.')
+    # Do some compensating work
+```
 
 {{% alert title="Note" color="primary" %}}
-In the example above, `"Step1"`, `"Step2"`, `"MyCompensation"`, etc. represent workflow activities, which are functions in your code that actually implement the steps of the workflow. For brevity, these activity implementations are left out of this example.
+Workflow retry policies will be available in a future version of the Python SDK.
```
Comment (msfussell) [inline daprdocs/content/en/developing-applications/building-blocks/workflow/workflow-patterns.md:107]: Just have to remember that statements like this need to be removed when ready and sometimes these get lost in updates.

File: daprdocs/content/en/developing-applications/building-blocks/workflow/workflow-patterns.md (hunk 1 above)
Comment (cgillum) [reply daprdocs/content/en/developing-applications/building-blocks/workflow/workflow-patterns.md:59]: For Python, the code was concise enough that I went ahead and included everything - thus I moved this note to be just part of the .NET tab.

File: daprdocs/content/en/developing-applications/building-blocks/workflow/workflow-patterns.md (hunk 2 above)
Comment (cgillum) [reply daprdocs/content/en/developing-applications/building-blocks/workflow/workflow-patterns.md:107]: I will open an issue on GitHub to track this work and will call out updating this doc specifically.

**Review by msfussell — APPROVED** [review]
LGTM

File: daprdocs/content/en/developing-applications/building-blocks/workflow/workflow-patterns.md (hunk 2 above)
Comment (cgillum) [reply daprdocs/content/en/developing-applications/building-blocks/workflow/workflow-patterns.md:107]: Here's the issue tracking this: https://github.com/dapr/python-sdk/issues/576

**Pull request merged**